numba/cuda/cg.py

```diff
@@ -0,0 +1,62 @@
+from numba.core import types
+from numba.core.extending import overload, overload_method
+from numba.core.typing import signature
+from numba.cuda import nvvmutils
+from numba.cuda.extending import intrinsic
+from numba.cuda.types import grid_group, GridGroup as GridGroupClass
+
+
+class GridGroup:
+    """A cooperative group representing the entire grid"""
+
+    def sync() -> None:
+        """Synchronize this grid group"""
+
+
+def this_grid() -> GridGroup:
+    """Get the current grid group."""
+    return GridGroup()
```

Comment:
```suggestion
```
does this need a Python implementation?

Reply:
I thought it might be instructive for anyone using an IDE that "peeks" at the implementation.

Reply:
Thanks for clarifying, am happy to leave as-is.